mapper/tedge_mapper/src/main.rs

```diff
@@ -18,27 +39,82 @@ async fn main() -> Result<(), anyhow::Error> {
         .with_span_events(tracing_subscriber::fmt::format::FmtSpan::CLOSE)
         .init();
 
-    info!("{} starting!", APP_NAME);
+    match cloud_name {
+        "c8y" => {
+            let _flockfile = check_another_instance_is_running(APP_NAME_C8Y)?;
 
-    let _lockfile = match flockfile::Flockfile::new_lock(format!("{}.lock", APP_NAME)) {
-        Ok(file) => file,
-        Err(err) => {
-            error!("Another instance of {} is running.", APP_NAME);
-            return Err(err.into());
+            info!("{} starting!", APP_NAME_C8Y);
+
+            let mqtt_config = mqtt_client::Config::default();
+            let mqtt = Client::connect(APP_NAME_C8Y, &mqtt_config).await?;
+
+            mapper::Mapper::new(
+                mqtt,
+                c8y_mapper::CumulocityMapperConfig::default(),
+                Box::new(c8y_mapper::CumulocityConverter),
+            )
+            .run()
+            .instrument(debug_span!(APP_NAME_C8Y))
+            .await?
         }
-    };
+        "az" => {
+            let _flockfile = check_another_instance_is_running(APP_NAME_AZ)?;
 
-    let config = mqtt_client::Config::default();
-    let mqtt = Client::connect(APP_NAME, &config).await?;
+            info!("{} starting!", APP_NAME_AZ);
 
-    let mapper = mapper::Mapper::new_from_string(
-        mqtt,
-        mapper::IN_TOPIC,
-        mapper::C8Y_TOPIC_C8Y_JSON,
-        mapper::ERRORS_TOPIC,
-    )?;
+            let mqtt_config = mqtt_client::Config::default();
+            let mqtt = Client::connect(APP_NAME_AZ, &mqtt_config).await?;
 
-    mapper.run().instrument(debug_span!(APP_NAME)).await?;
+            let config_repository = get_config_repository()?;
+            let tedge_config = config_repository.load()?;
+            mapper::Mapper::new(
+                mqtt,
+                az_mapper::AzureMapperConfig::default(),
+                Box::new(az_mapper::AzureConverter {
+                    add_timestamp: tedge_config.query(AzureMapperTimestamp)?.is_set(),
+                    time_provider: Box::new(SystemTimeProvider),
+                }),
+            )
+            .run()
+            .instrument(debug_span!(APP_NAME_AZ))
+            .await?
+        }
+        _ => return Err(MapperError::IncorrectArgument),
```

Comment:
If you have a `CloudName` enum, you can get rid of this case as you already handle it during parsing.